eth-types/src/lib.rs

```diff
@@ -376,52 +376,29 @@ pub struct GethExecTrace {
     pub struct_logs: Vec<GethExecStep>,
 }
 
-/// Truncate the memory in each step to the memory size before the step is
-/// executed (and before the memory is expanded).  This is required because geth
-/// sets the memory in each step as the memory before execution but after
-/// expansion.
-pub fn fix_geth_trace_memory_size(trace: &mut [GethExecStep]) {
-    let mut mem_sizes = vec![0; trace.len()];
-    let mut call_mem_size_stack = Vec::new();
-    for i in 1..trace.len() {
-        let step_prev = &trace[i - 1];
-        let step = &trace[i];
-        mem_sizes[i] = match step.depth as isize - step_prev.depth as isize {
-            // Same call context
-            0 => step_prev.memory.0.len(),
-            // into new call context
-            1 => {
-                call_mem_size_stack.push(step_prev.memory.0.len());
-                0
-            }
-            // return from call context
-            -1 => call_mem_size_stack.pop().expect("call stack is empty"),
-            _ => unreachable!(),
-        };
-    }
-    for i in 0..trace.len() {
-        trace[i].memory.0.truncate(mem_sizes[i]);
-    }
-}
-
-impl<'de> Deserialize<'de> for GethExecTrace {
-    fn deserialize<D>(deserializer: D) -> Result<GethExecTrace, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
+impl From<GethExecTraceInternal> for GethExecTrace {
```

Comment:
Now that we don't need to fix anything in the exec trace, the type `GethExecTraceInternal` is no longer necessary.
We just need to update `GethExecTrace` like this:
```diff
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Deserialize, Clone, Debug, Eq, PartialEq)]
 pub struct GethExecTrace {
     /// Used gas
     pub gas: Gas,
     /// True when the transaction has failed.
     pub failed: bool,
     /// Return value of execution
+    #[serde(rename = "returnValue")]
     pub return_value: String,
     /// Vector of geth execution steps of the trace.
+    #[serde(rename = "structLogs")]
     pub struct_logs: Vec<GethExecStep>,
 }
```

And remove the type `GethExecTraceInternal`, the traits impl `From<GethExecTraceInternal> for GethExecTrace` and `Deserialize<'de> for GethExecTrace`.
You'll need to update the `deserialize_geth_exec_trace2` test to use `GethExecTrace`